packages/migrate/src/__tests__/__helpers__/conditionalTests.ts

```diff
@@ -0,0 +1,166 @@
+import { Provider } from '@prisma/driver-adapter-utils'
+
+import { currentDriverAdapterName, DriverAdapterName, providerOfCurrentDriverAdapter } from './driverAdapters'
+
+// We have some tests for providers that are not supported by driver adapters yet.
+// TODO: D1 is also a special case as its tests have to be transformed to driver adapter tests.
+export type SupportedProviders = Provider | 'mongodb' | 'cockroachdb' | 'sqlserver' | 'd1'
+
+export type Matrix = {
+  providers: {
+    [K in SupportedProviders]?: boolean
+  }
+  driverAdapters: {
+    [K in DriverAdapterName]?: boolean
+  }
+  onlyDriverAdapters?: boolean
+}
+
+export const allProviders = {
+  mysql: true,
+  postgres: true,
+  sqlite: true,
+  mongodb: true,
+  cockroachdb: true,
+  sqlserver: true,
+  d1: true,
+}
+
+// Ensure type wise that `allProviders` contains all supported providers
+declare const __testAllProvidersA: keyof typeof allProviders
+declare const __testAllProvidersB: SupportedProviders
+// eslint-disable-next-line no-constant-condition
+if (false) {
+  __testAllProvidersB satisfies keyof typeof allProviders
+  __testAllProvidersA satisfies SupportedProviders
+}
+
+export const allDriverAdapters = {
+  pg: true,
+  'pg-worker': true,
+  planetscale: true,
+  neon: true,
+  libsql: true,
+  d1: true,
+}
+
+// Ensure type wise that `allDriverAdapters` actually contains all supported driver adapters
+declare const __testAllDriverAdaptersA: keyof typeof allDriverAdapters
+declare const __testAllDriverAdaptersB: DriverAdapterName
+// eslint-disable-next-line no-constant-condition
+if (false) {
+  __testAllDriverAdaptersB satisfies keyof typeof allDriverAdapters
+  __testAllDriverAdaptersA satisfies DriverAdapterName
+}
```

Comment:
This pattern with type tests is an overkill here, you can achieve the same thing easier:
```suggestion
export const allProviders = {
  mysql: true,
  postgres: true,
  sqlite: true,
  mongodb: true,
  cockroachdb: true,
  sqlserver: true,
  d1: true,
} satisfies Record<SupportedProviders, true>

export const allDriverAdapters = {
  pg: true,
  'pg-worker': true,
  planetscale: true,
  neon: true,
  libsql: true,
  d1: true,
} satisfies Record<DriverAdapterName, true>
```